### ISSUE TITLE
Vendor files missing stuff again

### ISSUE THREAD

**Issue by ToeiRei** [crowdsecurity/crowdsec#2413]
### What happened?

go: ariga.io/atlas@v0.7.2-0.20220927111110-867ee0cca56a: Get "https://proxy.golang.org/ariga.io/atlas/@v/v0.7.2-0.20220927111110-867ee0cca56a.mod": dial tcp [2a00:1450:4001:808::2011]:443: connect: network is unreachable
 * ERROR: net-analyzer/crowdsec-1.5.2::crowdsec-overlay failed (unpack phase):

### What did you expect to happen?

all required files in the vendor package

### How can we reproduce it (as minimally and precisely as possible)?

see https://github.com/ToeiRei/crowdsec-overlay/issues/19

### Crowdsec version

<details>

```console
$ cscli version
# paste output here
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Enabled collections and parsers

<details>

```console
$ cscli hub list -o raw
# paste output here
```

</details>

### Acquisition config

<details>
```console
# On Linux:
$ cat /etc/crowdsec/acquis.yaml /etc/crowdsec/acquis.d/*
# paste output here

# On Windows:
C:\> Get-Content C:\ProgramData\CrowdSec\config\acquis.yaml
# paste output here
</details>


### Config show

<details>

```console
$ cscli config show
# paste output here
```

</details>

### Prometheus metrics

<details>

```console
$ cscli metrics
# paste output here
```

</details>

### Related custom configs versions (if applicable) : notification plugins, custom scenarios, parsers etc.

<details>

</details>

**Comment by LaurenceJJones** [comment]
Odd, the timestamps should all be the same

```
[crowdsec-release.tgz](https://github.com/crowdsecurity/crowdsec/releases/download/v1.5.2/crowdsec-release.tgz)
47.4 MB
May 26
[crowdsec.1.5.2.nupkg](https://github.com/crowdsecurity/crowdsec/releases/download/v1.5.2/crowdsec.1.5.2.nupkg)
47.1 MB
May 26
[crowdsec_1.5.2.msi](https://github.com/crowdsecurity/crowdsec/releases/download/v1.5.2/crowdsec_1.5.2.msi)
47.7 MB
May 26
[vendor.tgz](https://github.com/crowdsecurity/crowdsec/releases/download/v1.5.2/vendor.tgz)
23.1 MB
Jun 17
Source code
(zip)
May 26
Source code
(tar.gz)
May 26
```

Ill check with @mmetc

**Comment by ToeiRei** [comment]
could also be that the vendor files are packaged wrong.
https://wiki.gentoo.org/wiki/Writing_go_Ebuilds is basically the preferred way

**Comment by mmetc** [comment]
Hi!

I added the vendor.tgz to the release by hand because, the CI integration to create and upload it was written after 1.5.2.

I was not aware that gentoo had a standard way to do that, so I just made a tarball. We can change it.

**Comment by ToeiRei** [comment]
That would be lovely. That's the preferred way for any go application as far as I know

**Comment by mmetc** [comment]
I don't have a gentoo machine at hand, the method we use is "vendor tarball" but I'm not sure of the name to use, I suppose crowdsec-{version}-vendor.tar.xz ? Since it contains vendor directories for the notification plugins too, is it extracted correctly?

**Comment by ToeiRei** [comment]
I do rename them in the ebuild internally as seen in https://github.com/ToeiRei/crowdsec-overlay/blob/b36456176de112a8f43d050f77f53aa249c69274/net-analyzer/crowdsec/crowdsec-1.5.2.ebuild#L12

**Comment by ToeiRei** [comment]
Looks like it's building here. I'll commit and bump the guy who had problems on my bug and report back

**Comment by mmetc** [comment]
Thanks, I'll merge for 1.5.3 then

**Comment by ToeiRei** [comment]
can you update the go-bouncers as well, please?

**Comment by ToeiRei** [comment]
My tester here has the same problem. I'm trying to get him on discord so we can figure it out together as it looks like the go packages are cached on my end.

**Comment by mmetc** [comment]
Hi @ToeiRei , sorry for the delay

Yes I did the change in the bouncers, you can see it in https://github.com/crowdsecurity/cs-firewall-bouncer/releases/tag/v0.0.28-rc5

Looking at your ebuild, a few minor and one major note for the upcoming 1.5.3.

 - you can drop the dependency on jq, it's not needed to build anymore
 - replacing a ROOT variable with MYROOT. It has been removed.
 - I just landed a PR to master that moves all plugins to cmd/, so they use the same dependencies and vendor file as crowdsec. Basically change `doexe plugins/notifications/splunk/notification-splunk` to `doexe cmd/notification-splunk/notification-splunk`.
 - we have a new "sentinel" plugin

The other main change is that now we build against the C++ libre2-dev. A pure go solution with WASM is available as a fallback, but it increases load times for cscli as well which is not acceptable in most cases.

Building should be as simple as adding a dependency on pkgconfig and the library itself, any version will do. We distribute static builds in the .deb and .rpm but understand it can be an issue on some platforms, so a dynamic link and runtime dependency is ok.

We should release an -rc5 any day now with the above changes. Let me know if I can help further, thanks!

**Comment by ToeiRei** [comment]
@mmetc, you made my day. Seriously.
Looking forward to the 'stable' tag to update.

**Comment by ToeiRei** [comment]
Hi @mmetc,

I started updating the ebuilds asap and by now I am at least able to reproduce the error at hands on **one** of my machines while it builds fine on the others. Not sure what's exactly missing or how that atlas dependency isn't in the vendor files.... 

At least we have a starting point here.

**Comment by ToeiRei** [comment]
According to the documentation, we may need to add a dependency tarball as well. Looks like https://github.com/golang/go/issues/26366 to me.

```
user $GOMODCACHE="${PWD}"/go-mod go mod download -modcacherw -x
user $tar --create --auto-compress --file foo-1-deps.tar.xz go-mod
```

**Comment by mmetc** [comment]
Hi @ToeiRei ,

if you scroll to the right in your message there is a network error:


```
go: ariga.io/atlas@v0.7.2-0.20220927111110-867ee0cca56a: Get "https://proxy.golang.org/ariga.io/atlas/@v/v0.7.2-                                                  0.20220927111110-867ee0cca56a.mod":
dial tcp: lookup proxy.golang.org on 1.1.1.1:53: dial udp 1.1.1.1:                                                  53: connect: network is unreachable

```

As for the mod cache - you already use the vendor directory, same purpose with another tool.

**Comment by ToeiRei** [comment]
The network error is intended. It is sandboxing.

**Comment by mmetc** [comment]
Of course, sorry for the confusion. This has nothing to do with atlas because it's simply the first line of go.sum. Either the vendor file is not created correctly, or it's not extracted in the right place.

I verified vendor.tgz in a sandboxed environment already (freebsd/poudriere) so the content of the .tar.xz should be correct too, they have the same files.

I'd rather avoid shipping the go-mod directory because if you notice, it includes EVERY source file in the dependent packages, not just the ones that are imported by crowdsec. It's 300MB vs 9MB.

**Comment by ToeiRei** [comment]
The vendor archive is incomplete due to a 'bug' or 'feature' on the go side. They extended the documentation on the gentoo side to that - see https://github.com/crowdsecurity/crowdsec/issues/2413#issuecomment-1728500695

**Comment by mmetc** [comment]
Hi, I can build 1.5.4 with this change

https://github.com/ToeiRei/crowdsec-overlay/pull/20

thanks and let me know if you need anything else!
